### PR TITLE
expose ExceptionClass only in debug mode

### PR DESCRIPTION
**Pull request by Yuqi-Du**
**What this PR does**:
expose ExceptionClass in error msg only in debug mode

**Which issue(s) this PR fixes**:
Fixes https://github.com/stargate/jsonapi/issues/578

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
Let's talk about this on daily meeting -- I am not sure if log level is a good determination for debug mode.

**Comment by jeffreyscarpenter** [comment]
> Let's talk about this on daily meeting -- I am not sure if log level is a good determination for debug mode.

I agree. Let's discuss making a property to control this

File: src/main/java/io/stargate/sgv2/jsonapi/exception/JsonApiException.java [hunk 1]
```diff
@@ -58,8 +60,17 @@ public CommandResult get() {
   }
 
   public CommandResult.Error getCommandResultError(String message) {
-    Map<String, Object> fields =
-        Map.of("errorCode", errorCode.name(), "exceptionClass", this.getClass().getSimpleName());
+    Map<String, Object> fields = null;
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/JsonApiException.java:63]: No need to assign here, assigned in all branches.

File: src/main/java/io/stargate/sgv2/jsonapi/exception/JsonApiException.java [hunk 2]
```diff
@@ -58,8 +60,17 @@ public CommandResult get() {
...
+    SmallRyeConfig config = ConfigProvider.getConfig().unwrap(SmallRyeConfig.class);
+    // enable debug mode for unit tests, since it can not be injected
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/JsonApiException.java:65]: I'm not sure I understand this... is there something special wrt unit tests?
Or do you mean that we cannot use injection here and need to explicitly access configuration this way?

File: src/main/java/io/stargate/sgv2/jsonapi/exception/DebugModeConfig.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package io.stargate.sgv2.jsonapi.exception;
+
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+@ConfigMapping(prefix = "stargate.debug")
+public interface DebugModeConfig {
+
+  @WithDefault("false")
+  boolean enable();
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/DebugModeConfig.java:10]: We may want to rename this as `enabled` to be compatible with other settings; I'll change.

**Review by tatu-at-datastax — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/exception/DebugModeConfig.java (hunk 3 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/exception/DebugModeConfig.java]: Renamed.

**Pull request merged**